contrib/inventory/ec2.py

```diff
@@ -338,8 +338,8 @@ def read_settings(self):
         else:
             self.all_elasticache_nodes = False
 
-        # boto configuration profile (prefer CLI argument)
-        self.boto_profile = self.args.boto_profile
+        # boto configuration profile (prefer environment variables then CLI argument)
+        self.boto_profile = os.environ.get('AWS_PROFILE', self.args.boto_profile)
```

Comment:
This gives precedence on selecting the profile to the environment over the command-line option. I'd prefer if this were the other way around so the precedence order (highest to lowest) would be:

1. Command-line flag
2. Environment variable
3. Config file